plutus-contract/test/Spec/Contract/Tx/Constraints/RequiredSigner.hs

```diff
@@ -156,3 +169,98 @@ mustBeSignedByTypedValidator = Scripts.mkTypedValidator @UnitTest
     $$(PlutusTx.compile [|| wrap ||])
     where
         wrap = Scripts.mkUntypedValidator
+
+
+-- plutus-tx-constraints tests
+-- all below to be covered by the above tests when the corresponding constraints will be implemented
+-- for CardanoTx
+
+
+{-# INLINABLE mustReferenceOutputValidatorV2 #-}
+mustReferenceOutputValidatorV2 :: PV2.Validator
+mustReferenceOutputValidatorV2 = PV2.mkValidatorScript
+    $$(PlutusTx.compile [|| wrap ||])
+ where
+     mkMustReferenceOutputV2Validator = mkMustReferenceOutputValidator Cons.V2.checkScriptContext
+     wrap = PSU.V2.mkUntypedValidator mkMustReferenceOutputV2Validator
```

Comment:
This test module covers requiredSigner, not reference output.

Reply:
Yes, I reused an existing script for TxConstraints and quickly add the offchain logic for offchain required signatures to it. I remove the validator and use the one of `Ledger.Test` instead.

Reply:
... because "required signer" is not the same as signatory witness for ledger spending condition. The only way to know is by inspecting the txbody or onchain validation.

Reply:
Ok, it should be fixed.

Reply:
Nevermind, I was stupid as it's pretty straightforward to adapt the initial contract for cardano constraints. Done.